File: src/main/java/com/smu/householdaccount/entity/Member.java
package com.smu.householdaccount.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;

import java.time.Instant;
import java.time.LocalDateTime;
import java.util.LinkedHashSet;
import java.util.Set;

@Getter
@Setter
@Entity
@Table(name = "MEMBER")
public class Member {
    @Id
    @Size(max = 50)
    @Column(name = "MEMBER_ID", nullable = false, length = 50)
    private String memberId;

    @Size(max = 255)
    @NotNull
    @Column(name = "PASSWORD", nullable = false)
    private String password;

    @Size(max = 20)
    @NotNull
    @Column(name = "ROLE", nullable = false, length = 20)
    private String role;

    @Size(max = 100)
    @NotNull
    @Column(name = "MEMBER_NAME", nullable = false, length = 100)
    private String memberName;

    @Size(max = 50)
    @NotNull
    @Column(name = "MEMBER_NICKNAME", nullable = false, length = 50)
    private String memberNickname;

    @Size(max = 255)
    @Column(name = "ADDRESS")
    private String address;

    @Size(max = 20)
    @Column(name = "PHONE", length = 20)
    private String phone;

    @ColumnDefault("'Y'")
    @Column(name = "ENABLED")
    private String enabled;


    @ColumnDefault("SYSTIMESTAMP")
    @Column(name = "CREATED_AT")
    private LocalDateTime createdAt;

    @Column(name = "UPDATED_AT")
    private LocalDateTime updatedAt;

//   @PrePersist, @PreUpdate 두개추가 회원가입시 자동 createdAt 넣어줌, 정보수정시 자동 updatedAt
    @PrePersist
    public void prePersist() {
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
        if (this.enabled == null) {
            this.enabled = "Y";
        }
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }




    @OneToMany(mappedBy = "writer")
    private Set<BoardComment> boardComments = new LinkedHashSet<>();

    @OneToMany(mappedBy = "member")
    private Set<BoardLike> boardLikes = new LinkedHashSet<>();

    @OneToMany(mappedBy = "writer")
    private Set<BoardPost> boardPosts = new LinkedHashSet<>();

    @OneToMany(mappedBy = "owner")
    private Set<BudgetGroup> budgetGroups = new LinkedHashSet<>();

    @OneToMany(mappedBy = "member")
    private Set<GroupMember> groupMembers = new LinkedHashSet<>();

    @OneToMany(mappedBy = "member")
    private Set<ItemWish> itemWishes = new LinkedHashSet<>();

    @OneToMany(mappedBy = "member")
    private Set<LedgerEntry> ledgerEntries = new LinkedHashSet<>();

    @OneToMany(mappedBy = "buyer")
    private Set<OrderMain> orderMains = new LinkedHashSet<>();

    @OneToMany(mappedBy = "member")
    private Set<PaymentTransaction> paymentTransactions = new LinkedHashSet<>();

    @OneToMany(mappedBy = "member")
    private Set<Seller> sellers = new LinkedHashSet<>();

}

File: src/main/java/com/smu/householdaccount/entity/Member.kt
package com.smu.householdaccount.entity

import jakarta.persistence.*
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size
import org.hibernate.annotations.ColumnDefault
import java.time.Instant

@Entity
@Table(name = "MEMBER")
open class Member {
    @Id
    @Size(max = 50)
    @Column(name = "MEMBER_ID", nullable = false, length = 50)
    open var memberId: String? = null

    @Size(max = 255)
    @NotNull
    @Column(name = "PASSWORD", nullable = false)
    open var password: String? = null

    @Size(max = 20)
    @NotNull
    @Column(name = "ROLE", nullable = false, length = 20)
    open var role: String? = null

    @Size(max = 100)
    @NotNull
    @Column(name = "MEMBER_NAME", nullable = false, length = 100)
    open var memberName: String? = null

    @Size(max = 50)
    @NotNull
    @Column(name = "MEMBER_NICKNAME", nullable = false, length = 50)
    open var memberNickname: String? = null

    @Size(max = 255)
    @Column(name = "ADDRESS")
    open var address: String? = null

    @Size(max = 20)
    @Column(name = "PHONE", length = 20)
    open var phone: String? = null

    @ColumnDefault("'Y'")
    @Column(name = "ENABLED")
    open var enabled: Boolean? = null

    @ColumnDefault("SYSTIMESTAMP")
    @Column(name = "CREATED_AT")
    open var createdAt: Instant? = null

    @Column(name = "UPDATED_AT")
    open var updatedAt: Instant? = null

    @OneToMany(mappedBy = "writer")
    open var boardComments: MutableSet<BoardComment> = mutableSetOf()

    @OneToMany(mappedBy = "member")
    open var boardLikes: MutableSet<BoardLike> = mutableSetOf()

    @OneToMany(mappedBy = "writer")
    open var boardPosts: MutableSet<BoardPost> = mutableSetOf()

    @OneToMany(mappedBy = "owner")
    open var budgetGroups: MutableSet<BudgetGroup> = mutableSetOf()

    @OneToMany(mappedBy = "member")
    open var groupMembers: MutableSet<GroupMember> = mutableSetOf()

    @OneToMany(mappedBy = "member")
    open var ledgerEntries: MutableSet<LedgerEntry> = mutableSetOf()

    @OneToMany(mappedBy = "buyer")
    open var orderMains: MutableSet<OrderMain> = mutableSetOf()

    @OneToMany(mappedBy = "member")
    open var sellers: MutableSet<Seller> = mutableSetOf()
}